src/webview.cpp

```diff
@@ -166,12 +166,13 @@ QWebEngineView* WebView::createWindow(QWebEnginePage::WebWindowType type)
 
 void WebView::onUrlChanged(const QUrl& url) {
     auto zimId = getZimIdFromUrl(url);
+    auto app = KiwixApp::instance();
+    app->saveListOfOpenTabs();
```

Comment:
I replaced the previous call on `webpage.cpp` as I found that was a bad choice. No need for the `acceptNavigationRequest`. onUrlChanged is indeed called on in-page URL navigations.